### ISSUE TITLE
Giving coalition medals/commendations to engineers with electrochemist perk bug

### ISSUE THREAD

**Issue by bryson099** [Regalis11/Barotrauma#7210]
-I have searched the issue tracker to check if the issue has already been reported.

**Description**
Giving an engineer with electrochemist perk enough medals to level up will roll back all their XP and they will lose their talent points. They will also be prevented from gaining talent points in the future until they create a new character in a campaign. It will still seem like the engineer can choose which talents to acquire but the reset/apply button will be greyed out, and after they close the menu it will reset the talents they have chosen. 

**Steps To Reproduce**
Give an engineer with the electrochemist perk enough medals to level up once.

**Version**
Windows 10 Barotrauma v0.15.13.0

**Comment by doornydew** [comment]
already reported.

**Comment by bryson099** [comment]
My bad, you had reported it 1 minute earlier than I did. Eitherway, this bug does not exist because I looked into it a little bit more and it is actually related to the perk the "All-seeing-eye" since it gives you a ton of free perks without actually levelling up. The bug will fix itself after you level up by the same amount of levels it would've taken to acquire all the perks you got for free by unlocking the all-seeing-eye. Only thought there was a correlation with electrochemist and medals since electrochemist was obtained by unlocking the all-seeing-eye and medals were used to level up.
